Mathlib/Data/List/Basic.lean

```diff
@@ -771,6 +771,9 @@ theorem getLast?_append {l₁ l₂ : List α} {x : α} (h : x ∈ l₂.getLast?)
 
 /-! ### head(!?) and tail -/
 
+@[simp]
+theorem head_empty : List.head! [] = 0 := rfl
```

Comment:
```suggestion
theorem head_nil : List.head! [] = 0 := rfl
```
Sorry, I know that I was the one who named these lemmas, but I think that `nil` is the standard name, not `head`.